src/platform/silabs/tracing/SilabsTracing.cpp

```diff
@@ -0,0 +1,277 @@
+#include "SilabsTracing.h"
+#include <cstring>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/PersistentData.h>
+
+#if !CONFIG_BUILD_FOR_HOST_UNIT_TEST
+#include <Logging.h> // for isLogInitialized
+#endif
+
+// The following functions needs to be implemented by the application to allows logging or storing the traces /
+// metrics. If the application does not implement them, the traces will simply be ignored.
+bool __attribute__((weak)) isLogInitialized()
+{
+    return false;
+}
+
+namespace chip {
+namespace Tracing {
+namespace Silabs {
+
+// TODO add string mapping to log the operation names and types
+
+static constexpr size_t kPersistentTimeTrackerBufferMax = SERIALIZED_TIME_TRACKERS_SIZE_BYTES;
+
+struct TimeTrackerStorage : public TimeTracker, PersistentData<kPersistentTimeTrackerBufferMax>
+{
+    // TODO implement the Persistent Array class and use it here for logging the watermark array
+};
+
+SilabsTracer SilabsTracer::sInstance;
+
+SilabsTracer::SilabsTracer()
+{
+    VerifyOrDie(Init() == CHIP_NO_ERROR);
+}
+
+void SilabsTracer::TraceBufferClear()
+{
+    // Clear the time tracker list
+    while (mTimeTrackerList.head != nullptr)
+    {
+        mTimeTrackerList.Remove(0);
+    }
+    mBufferedTrackerCount = 0;
+}
+
+CHIP_ERROR SilabsTracer::Init()
+{
+    TraceBufferClear();
+
+    // Initialize the time trackers
+    memset(mLatestTimeTrackers, 0, sizeof(mLatestTimeTrackers));
+    memset(mWatermarks, 0, sizeof(mWatermarks));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SilabsTracer::StartWatermarksStorage(PersistentStorageDelegate * storage)
```

Comment:
```suggestion
CHIP_ERROR SilabsTracer::SetWatermarksStorage(PersistentStorageDelegate * storage)
```
Start seems like an odd name

Reply:
Except this is exactly what it does, Set is misleading as it doesn't mean anything.